_posts/2019-09-27-Combining-MongoDB-and-AWS-Lambda.md

```diff
@@ -0,0 +1,272 @@
+* How to setup a production like structure
+* Comparing MongoDB performance with AWS native solutions
+
+# Connect your Lambda Functions with your MongoDB Atlas Cluster
+We want to deploy a production grade setup.  
+This means we won't connect over the open internet.
+We'll setup a VPC peering connection between our Atlas Cluster and our AWS VPC.  
+
+## The AWS side
+Let's setup a new VPC.
+In this VPC we will create a public subnet.  
+A route table will be associated with that subnet.
+In the route table we'll define that we want to route all database traffic through the VPC peering connection towards the Atlas cluster.
+
+In the AWS User Interface navigate to the VPC dashboard and click `Launch VPC Wizard`.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/1-VPC-dashboard.png" width="70%" height="70%">
+</div>
+
+Select that you want to create a VPC with a single public subnet.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/2-VPC-with-single-public-subnet.png" width="70%" height="70%">
+</div>
+
+Specify how big you want the IP range of this VPC to be.
+If you have trouble figuring out the relation between the `CIDR block` and the `Network Range` use one of the online converters to help you. ([https://www.ipaddressguide.com/cidr](https://www.ipaddressguide.com/cidr))  
+Give your VPC and public subnet a name.  
+Make sure that `enable DNS hostnames` is enabled.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/3-VPC-with-single-public-subnet-2.png" width="100%" height="100%">
+</div>
+
+Your VPC has been successfully created.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/4-VPC-successfully-created.png" width="70%" height="70%">
+</div>
+
+You will now notice that a new subnet and two new route tables have been created.  
+That is a route table for your VPC and a route table specifically for you public subnet.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/5-new-subnet.png" width="70%" height="70%">
+</div>
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/6-new-route-tables.png" width="70%" height="70%">
+</div>
+
+Before we go to MongoDB Atlas get some specific data about your VPC:
+* Write down the VPC-id
+* Write down the IPv4 CIDR of your VPC
+* Write down the security group associated with this vpc.
+Under security groups fetch the id of the security group that is associated with your vpc.
+
+<div style="text-align: center;" >
+  <img src="/img/2019-09-27-Combining-MongoDB-Atlas-and-AWS-Lambda/7-security-group.png" width="70%" height="70%">
+</div>
+
+
+## The MongoDB side
+
+Setup the MongoDB cluster.
+This has to be a dedicated cluster which means you'll need at least an M10.
```

Comment:
Hmm, is it because:
When you get a dedicated cluster that means you get a dedicated server on which your MongoDB database is hosted and thus peering and routing can be setup on that server?

Reply:
Probably that is true, but just mention what's in the docs ;-)